openedx/core/djangoapps/content_libraries/library_context.py

```diff
@@ -30,47 +32,63 @@ def __init__(self, **kwargs):
 
-        usage_key: the UsageKeyV2 subclass used for this learning context
-
-        Must return a boolean.
+        May raise ContentLibraryNotFound if the library does not exist.
         """
+        assert isinstance(usage_key, LibraryUsageLocatorV2)
         try:
             api.require_permission_for_library_key(usage_key.lib_key, user, permissions.CAN_EDIT_THIS_CONTENT_LIBRARY)
-        except (PermissionDenied, api.ContentLibraryNotFound):
+            return True
+        except PermissionDenied:
             return False
+        except api.ContentLibraryNotFound as exc:
+            # A 404 is probably what you want in this case, not a 500 error, so do that by default.
+            raise NotFound(f"Content Library '{usage_key.lib_key}' does not exist") from exc
+
+    def can_view_block_for_editing(self, user: UserType, usage_key: UsageKeyV2) -> bool:
+        """
+        Assuming a block with the specified ID (usage_key) exists, does the
+        specified user have permission to view its fields and OLX details (but
+        not necessarily to make changes to it)?
 
-        return self.block_exists(usage_key)
+        May raise ContentLibraryNotFound if the library does not exist.
+        """
+        assert isinstance(usage_key, LibraryUsageLocatorV2)
+        try:
+            api.require_permission_for_library_key(usage_key.lib_key, user, permissions.CAN_VIEW_THIS_CONTENT_LIBRARY)
+            return True
+        except PermissionDenied:
+            return False
+        except api.ContentLibraryNotFound as exc:
+            # A 404 is probably what you want in this case, not a 500 error, so do that by default.
+            raise NotFound(f"Content Library '{usage_key.lib_key}' does not exist") from exc
 
-    def can_view_block(self, user, usage_key):
+    def can_view_block(self, user: UserType, usage_key: UsageKeyV2) -> bool:
         """
         Does the specified usage key exist in its context, and if so, does the
         specified user have permission to view it and interact with it (call
         handlers, save user state, etc.)?
 
-        user: a Django User object (may be an AnonymousUser)
-
-        usage_key: the UsageKeyV2 subclass used for this learning context
-
-        Must return a boolean.
+        May raise ContentLibraryNotFound if the library does not exist.
         """
+        assert isinstance(usage_key, LibraryUsageLocatorV2)
         try:
             api.require_permission_for_library_key(
                 usage_key.lib_key, user, permissions.CAN_LEARN_FROM_THIS_CONTENT_LIBRARY,
             )
-        except (PermissionDenied, api.ContentLibraryNotFound):
+            return True
+        except PermissionDenied:
             return False
+        except api.ContentLibraryNotFound as exc:
+            # A 404 is probably what you want in this case, not a 500 error, so do that by default.
+            raise NotFound(f"Content Library '{usage_key.lib_key}' does not exist") from exc
```

Comment:
Should `can_view_block`, `can_view_block_for_editing`, and `can_edit_block` all call the same underlying helper with different permissions arguments?

Reply:
> Should `can_view_block`, `can_view_block_for_editing`, and `can_edit_block` all call the same underlying helper with different permissions arguments?

Yes of course they should 🤦🏻‍♂️. Done.